### PR TITLE
Feature/multiple allow list

### PR DESCRIPTION
**Pull request by piyalbasu**
https://github.com/stellar/freighter/assets/6789586/2c6a91fe-e90c-4721-b325-257c72b46b7d

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ module.exports = {
     "react/jsx-filename-extension": ["error", { extensions: [".tsx", ".jsx"] }],
     "jsdoc/newline-after-description": "off",
     "max-len": "off",
+    "no-await-in-loop": "off",
```
Comment (piyalbasu) [inline .eslintrc.js:34]: we can get rid of this, right? it'll be one last lint rule to ignore in the code lol

File: @shared/api/internal.ts [hunk 2]
```diff
@@ -1162,7 +1163,9 @@ export const editCustomNetwork = async ({
   return response;
 };
 
-export const loadSettings = (): Promise<Settings & IndexerSettings> =>
+export const loadSettings = (): Promise<
+  Settings & IndexerSettings & { assetsLists: AssetsLists }
```
Comment (piyalbasu) [inline @shared/api/internal.ts:1167]: assets lists don't quite fall into the paradigm of general app settings or settings from the indexer. Maybe if this "3rd thing" category grows, we can add a separate service to deal with it

File: @shared/api/types.ts [hunk 3]
```diff
@@ -66,7 +68,7 @@ export interface Response {
   network: string;
   networkIndex: number;
   networkName: string;
-  recentAddresses: Array<string>;
+  recentAddresses: string[];
```
Comment (piyalbasu) [inline @shared/api/types.ts:71]: fixing a couple linting warnings while I'm in here

File: extension/src/background/helpers/dataStorage.ts [hunk 4]
```diff
@@ -254,6 +256,18 @@ export const resetAccountSubscriptions = async () => {
     // once account is unlocked, setup Mercury account subscription if !HAS_ACCOUNT_SUBSCRIPTION
     await localStore.setItem(HAS_ACCOUNT_SUBSCRIPTION, {});
   }
+  await migrateDataStorageVersion("4.0.2");
```
Comment (piyalbasu) [inline extension/src/background/helpers/dataStorage.ts:259]: the `resetAccountSubscription` migration forgot to migrate the version, so adding this in here now

File: .eslintrc.js (hunk 1 above)
Comment (aristidesstaffieri) [reply .eslintrc.js:34]: 😄 yes please

File: @shared/constants/soroban/token.ts [hunk 5]
```diff
@@ -26,3 +28,35 @@ export interface SorobanToken {
...
+};
+
+export interface AssetsListItem {
+  url: string;
+  enabled: boolean;
+}
+
+export const DEFAULT_ASSETS_LISTS: AssetsLists = {
+  [NETWORKS.PUBLIC]: [
+    {
```
Comment (aristidesstaffieri) [inline @shared/constants/soroban/token.ts:45]: In the mocks, it looks like we will allow deleting a list but always want to retain one list as the "default" or one list that the user cannot delete. Should we add a field to the stellar expert list in order to key that one as the default one?

File: extension/src/background/helpers/dataStorage.ts (hunk 4 above)
Comment (aristidesstaffieri) [reply extension/src/background/helpers/dataStorage.ts:259]: omg, thank you

File: extension/src/popup/views/ManageAssetsLists/index.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            </ListNavLinkWrapper>
+          </div>
+        )}
+      </View.Content>
+      <View.Footer>
+        <Button
+          size="md"
+          isFullWidth
+          variant="tertiary"
+          onClick={() => navigateTo(ROUTES.account)}
```
Comment (aristidesstaffieri) [inline extension/src/popup/views/ManageAssetsLists/index.tsx:113]: assuming the add list is "TODO"

**Review by aristidesstaffieri — APPROVED** [review]

File: extension/src/popup/views/ManageAssetsLists/index.tsx (hunk 6 above)
Comment (piyalbasu) [reply extension/src/popup/views/ManageAssetsLists/index.tsx:113]: Yup, just placeholder until the next view is created

File: @shared/constants/soroban/token.ts (hunk 5 above)
Comment (piyalbasu) [reply @shared/constants/soroban/token.ts:45]: Yeah, good question. I need to follow up with product on this. Not sure which of these should be "undeletable"

**Pull request merged**